### DISCUSSION TITLE
Encrypted dash stream: I have the key

### DISCUSSION THREAD

**Discussion by Chris230291** [streamlink/streamlink#5387] [Q&A] [answered]
### Description

I would like to stream an encrypted dash stream and pipe the output to stdout. I saw mention of a key file... Is there a way to pass the key directly in the command? Am I correct to assume if I use `best` it will take only the the best quality video track but still include all other tracks, such as languages and subtitles when present?

Thanks

**Comment by bastimeyer** [comment, marked as answer]
This is unsupported. There's a fork of Streamlink which removes the encryption detection and which allows passing a key to the FFMpeg command which Streamlink spawns. 
https://github.com/streamlink/streamlink/discussions/5123#discussioncomment-4796568

----

`best` is what is considered the best stream quality of all returned streams.
https://streamlink.github.io/cli/tutorial.html#getting-started

The streams returned from the DASHStream implementation is the cartesian product of all available video and audio streams of the selected DASH period, minus duplicates when it's the same bitrate or resolution. This selection mechanism isn't ideal. See #4902
